plots and metrics: update new targets rules (#1809)
iterative/dvc.org: content/docs/command-reference/metrics/diff.md

## Options

- `--targets <paths>` - limit command scope to these metric files (supports any

>   file, even when not found as `metrics` in `dvc.yaml`). Using `-R`, directories

Reviewer: I think we should probably avoid "supports any file, even when not found as `metrics` in `dvc.yaml`" everywhere. It's better to specify at the top level that plots files are by definition could be anywhere. And `dvc.yaml` is used to get defaults (among other things)

>   to search metric files in can also be given. When specifying arguments for

Reviewer 2: ```suggestion
  to search for metrics files can also be given. When specifying arguments for
```

>   to search for metrics files can also be given. When specifying arguments for
>   `--targets` before `revisions`, you should use `--` after this option's
>   arguments, e.g.:

Reviewer 2: I think we should separate these paragraphs at this point. I'll do it:

```suggestion
  to search for metrics files can also be given.
  
  When specifying arguments for `--targets` before `revisions`, you should use
  `--` after this option's arguments, e.g.:
```

## Options

> - `--targets <paths>` - limit command scope to these metrics files (supports any
>   file, even when not found as `metrics` in `dvc.yaml`). Using `-R`, directories
>   to search for metrics files can also be given.

Reviewer 2: ```suggestion
- `--targets <paths>` - limit command scope to these files (supports any file,
  even when not found as `metrics` in `dvc.yaml`). Using `-R`, directories to
  search for metrics files can also be given.
```

> - `--targets <paths>` - limit command scope to these metrics files (supports any
>   file, even when not found as `metrics` in `dvc.yaml`). Using `-R`, directories
>   to search for metrics files can also be given.

Reviewer 2: ```suggestion
- `--targets <paths>` - limit command scope to these files (supports any valid
  metrics, even when not found as `metrics` in `dvc.yaml`). Using `-R`,
  directories to search for metrics files can also be given.
```

>   directories to search for metrics files can also be given.

Reviewer 2: ```suggestion
  directories to search for metrics can also be given.
```
